autogpt_platform/frontend/src/components/Flow.tsx

```diff
@@ -410,6 +410,31 @@ const FlowEditor: React.FC<{
 
   const { x, y, zoom } = useViewport();
 
+  // Set the initial view port to center the canvas.
+  useEffect(() => {
+    if (nodes.length <= 0 || x !== 0 || y !== 0) {
+      return;
+    }
+    const topLeft = { x: Infinity, y: Infinity };
+    const bottomRight = { x: -Infinity, y: -Infinity };
+    nodes.forEach((node) => {
+      const { x, y } = node.position;
+      topLeft.x = Math.min(topLeft.x, x);
+      topLeft.y = Math.min(topLeft.y, y);
+      bottomRight.x = Math.max(bottomRight.x, x + 100);
```

Comment:
Do these two mean that we're assuming nodes are 100 by 100?

Reply:
Yes, it's just some hardcoded assumption to avoid centering a single node graph tilted to the right

Reply:
I've added comments and fixed the calculation for auto-zooming feature.